src/comlink.ts

```diff
@@ -392,7 +396,24 @@ function closeEndPoint(endpoint: Endpoint) {
 }
 
 export function wrap<T>(ep: Endpoint, target?: any): Remote<T> {
-  return createProxy<T>(ep, [], target) as any;
+  const pendingListeners : PendingListenersMap = new Map();
+
+  ep.addEventListener("message", function handleMessage(ev: Event) {
+    const { data } = ev as MessageEvent;
+    if (!data || !data.id) {
+      return;
+    }
+    const resolver = pendingListeners.get(data.id);
+    if (resolver) {
```

Comment:
nit: flip this around to early return so that early return is the "no more work to do case"

```
if (!resolver) {
  return;
}

// actual work
...
```